src/plugins/intel_gpu/include/intel_gpu/primitives/range.hpp

```diff
@@ -10,9 +10,19 @@ namespace cldnn {
 struct range: public primitive_base<range> {
     CLDNN_DECLARE_PRIMITIVE(range)
 
-    range(const primitive_id &id, const std::vector<primitive_id> &input, const layout &output_layout) :
-        primitive_base { id, input, output_layout.data_padding }, output_layout { output_layout } {
-    }
-    layout output_layout;
+    /// @brief Constructs range primitive.
+    /// @param id This primitive id.
+    /// @param inputs Input primitive id vector.
+    /// @param out_pshape requested range output partial shape.
+    /// @param out_dtype  requested range output data_types.
+    range(const primitive_id &id,
+          const std::vector<primitive_id> &input,
+          const data_types out_dtype,
+          const ov::PartialShape &out_pshape = ov::PartialShape())
```

Comment:
I think we dont need out_pshape to be PartialShape, because it will not be used in the PartialShape use case in the future. 

So.. in calc_output_layouts:

if (memory_deps..) {
   // ngraph_shape_infer 
} else if (output_layout == valid) {
   // use output_layout
} else {
   return dynamic shape
}

Reply:
To do so, this part also should be changed, 
cldnn::tensor outTensor { cldnn::spatial(outShape[0]) }; to cldnn::batch(outShape[0]) at op creation of static shape usage

Reply:
I rollback the param from out_pshape to output_layout and change the calculation of work group size to set it regardless the output shape setting such as cldnn::tensor(batch(1)) and cldnn::tensor(sptial(1)).